crates/next-api/src/app.rs

```diff
@@ -862,7 +862,41 @@ impl AppEndpoint {
                 }
                 let client_shared_availability_info = client_shared_chunk_group.availability_info;
 
-                let client_references = client_reference_graph(Vc::cell(vec![rsc_entry_asset]));
+                let client_references = {
+                    let modules = if let Some(module_part_asset) =
+                        Vc::try_resolve_downcast_type::<EcmascriptModulePartAsset>(rsc_entry)
+                            .await?
+                    {
+                        let modules: Vec<_> = module_part_asset
+                            .await?
+                            .full_module
+                            .await?
+                            .inner_assets
+                            .iter()
+                            .try_join()
+                            .await?
+                            .iter()
+                            .flat_map(|a| a.values())
+                            .copied()
+                            .chain(std::iter::once(Vc::upcast(module_part_asset)))
+                            .collect();
+                        modules
+                    } else {
+                        vec![rsc_entry]
+                    };
+
+                    let mut client_references: ClientReferenceGraphResult =
+                        ClientReferenceGraphResult::default();
+
+                    for module in modules {
```

Comment:
I couldn't
- figure out how to do this without a `collect ` (how to unify the iterator types)
- figure out how to do this with less operations in the if-branch